pyo3-macros-backend/src/method.rs

```diff
@@ -128,12 +130,17 @@ impl<'a> FnSpec<'a> {
         } = parse_method_attributes(meth_attrs, allow_custom_name)?;
 
         let mut arguments = Vec::new();
-        let mut inputs_iter = sig.inputs.iter();
 
+        // TODO: maybe there's a cleaner solution
+        let inputs_empty = sig.inputs.is_empty();
+        let sig_span = sig.span();
+        let inputs_span = sig.inputs.span();
```

Comment:
Looks like this is down to lifetime of the `iter_mut()` borrow? Perhaps we can remove the need for these by rearranging the code.

Reply:
Yes, I couldn't leave them inline, because then it would be an immutable borrow during a mutable reference lifetime. I'll try to rearrange it somehow